### PR TITLE
AS-376 Add support for MfgTest

### PR DESCRIPTION
**Pull request by rabad-matternet**
- port changes mttr-cube-orange-mfg-testing branch for MfgTest
- changes are conditionally compiled for MttrCubeOrangeMfgTest
- added hwdef for MttrCubeOrangeMfgTest

### CODE REVIEW

**Comment by rabad-matternet** [comment]
**To all reviewers**, this PR is a port of the following two commits for MfgTest functionality:
- [f5a92ed](https://github.com/matternet/ardupilot/commit/f5a92ed27aff4c755377963f16f2d48e7427e926)
- [0x3867c](https://github.com/matternet/ardupilot/commit/093867c06eb6e348f421c7d11dbe807e4db09b44)

The port is done such that the changes are conditionally compiled and built with MttrCubeOrangeMfgTest (not MttrCubeOrange, which is used for production)

File: libraries/AP_HAL_ChibiOS/hwdef/MttrCubeOrangeMfgTest/defaults.parm
Comment (adriennet823) [inline libraries/AP_HAL_ChibiOS/hwdef/MttrCubeOrangeMfgTest/defaults.parm:1]: Is there a particular set of params you need for this test? If not, then I recommend aligning this with the latest D&R model.

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp [hunk 1]
```diff
@@ -445,11 +445,26 @@ void AP_RangeFinder_LightWareI2C::update(void)
 
 void AP_RangeFinder_LightWareI2C::sf20_timer(void)
 {
+#ifdef MFG_TEST_BUILD
+    static const uint32_t TIMER_FREQ = 20;
```
Comment (adriennet823) [inline libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp]: recommend adding units to this FREQ variable

**Review by adriennet823 — APPROVED** [review]

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp (hunk 1 above)
Comment (rabad-matternet) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp]: Done!

File: libraries/AP_HAL_ChibiOS/hwdef/MttrCubeOrangeMfgTest/defaults.parm
Comment (rabad-matternet) [reply libraries/AP_HAL_ChibiOS/hwdef/MttrCubeOrangeMfgTest/defaults.parm:1]: Discussed offline.  We will set these values to known values in the production (MttrCubeOrange) defaults.parm.

**Review by bhiln — APPROVED** [review]

**Pull request merged**